### PR TITLE
[fix](nereids) fix UnknownValue's reference in  simplify range rule

### PR DESCRIPTION
**Pull request by yujun777**
### What problem does this PR solve?

for sql 'a > 10 and b > 20' will generate an UnknowValue  with  sourceValues = [ Range(a > 10),  Range(b > 20) ].

in UnknowValue's constructor,  it set reference = sourceValues[0].reference, then its reference = a. This is an error.

when A union/interset B  generating an UnknowValue,  this UnknownValue's reference should be:
1. A.reference if A.reference = B.reference;
2. expr(A union/interset B)  if A.reference != B.reference;

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [x] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/SimplifyRange.java [hunk 1]
```diff
@@ -498,41 +494,53 @@ private UnknownValue(ExpressionRewriteContext context, Expression expr) {
...
-                List<ValueDesc> sourceValues, Expression exprForNonNull, BinaryOperator<Expression> mergeExprOp) {
-            super(context, sourceValues.get(0).reference, exprForNonNull);
+        public UnknownValue(ExpressionRewriteContext context, Expression toExpr,
+                List<ValueDesc> sourceValues, BinaryOperator<Expression> mergeExprOp) {
+            super(context, genReference(sourceValues, toExpr), toExpr);
             this.sourceValues = ImmutableList.copyOf(sourceValues);
             this.mergeExprOp = mergeExprOp;
         }
 
+        private static Expression genReference(List<ValueDesc> sourceValues, Expression toExpr) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/SimplifyRange.java:504]: add some comment to explain why need generate reference from `toExpr`

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/SimplifyRange.java [hunk 2]
```diff
@@ -271,8 +271,8 @@ public static ValueDesc discrete(ExpressionRewriteContext context, InPredicate i
 
     private static class EmptyValue extends ValueDesc {
 
-        public EmptyValue(ExpressionRewriteContext context, Expression reference, Expression exprForNonNull) {
-            super(context, reference, exprForNonNull);
+        public EmptyValue(ExpressionRewriteContext context, Expression reference, Expression toExpr) {
+            super(context, reference, toExpr);
         }
 
         @Override
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/SimplifyRange.java:278]: do not need to change `union` and `intersect` of `EmptyValue`?

Comment (yujun777) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/SimplifyRange.java:278]: for A union / intersect B,  if met A is EmptyValue,  then A and B's reference should always be the same. 

so EmptyValue union another ValueDesc with the same reference,  the EmptyValue really remove ? and insect with aother valueDesc with the same reference,  only keep this EmptyValue is ok ?

Comment (yujun777) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/SimplifyRange.java:278]: > do not need to change `union` and `intersect` of `EmptyValue`?

if A op B's references are different,  then generate a UnknownValue,  this UnknownValue would keep both A and B, no delete any of them.

so if A op B  got result is A, or result is B,   then we known that A and B's referece must be the same. 

so for A op B,  if A is EmptyValue, then
1) if B's reference different with A's reference,   then generate a UnknownValue,  UnknownValue's toExpression will keep both A and B
2) if B's reference equals to A's reference,  then EmptyValue union B = B is ok,   EmptyValue intersect B = EmptyValue is ok too.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/SimplifyRange.java (hunk 1 above)
Comment (yujun777) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/SimplifyRange.java]: > add some comment to explain why need generate reference from `toExpr`

had add comment

**Comment by yujun777** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**